pkg/analyzer/analyzers/opsgenie/opsgenie.go

```diff
@@ -14,8 +14,79 @@ import (
 	"github.com/jedib0t/go-pretty/table"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/analyzers"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/config"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/pb/analyzerpb"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
 )
 
+var _ analyzers.Analyzer = (*Analyzer)(nil)
+
+type Analyzer struct {
+	Cfg *config.Config
+}
+
+func (Analyzer) Type() analyzerpb.AnalyzerType { return analyzerpb.AnalyzerType_Opsgenie }
+
+func (a Analyzer) Analyze(_ context.Context, credInfo map[string]string) (*analyzers.AnalyzerResult, error) {
+	key, ok := credInfo["key"]
+	if !ok {
+		return nil, errors.New("missing key in credInfo")
+	}
+	info, err := AnalyzePermissions(a.Cfg, key)
+	if err != nil {
+		return nil, err
+	}
+	return secretInfoToAnalyzerResult(info), nil
+}
+
+func secretInfoToAnalyzerResult(info *SecretInfo) *analyzers.AnalyzerResult {
+	if info == nil {
+		return nil
+	}
+	result := analyzers.AnalyzerResult{
+		AnalyzerType:       analyzerpb.AnalyzerType_Opsgenie,
+		Metadata:           nil,
+		Bindings:           make([]analyzers.Binding, len(info.Permissions)),
+		UnboundedResources: make([]analyzers.Resource, len(info.Users)),
+	}
+
+	// Opsgenie has API integrations, so the key does not belong
+	// to a particular user or account, it itself is a resource
+	resource := analyzers.Resource{
+		Name:               "Opsgenie API Integration Key",
+		FullyQualifiedName: "Opsgenie API Integration Key",
+		Type:               "API Key",
+		Metadata: map[string]any{
+			"expires": "never",
+		},
+	}
+
+	for idx, permission := range info.Permissions {
+		result.Bindings[idx] = analyzers.Binding{
+			Resource: resource,
+			Permission: analyzers.Permission{
+				Value: permission,
+			},
+		}
+	}
+
+	// We can find list of users in the current account
+	// if the API key has Configuration Access, so these can be
+	// unbounded resources
+	for _, user := range info.Users {
+		result.UnboundedResources = append(result.UnboundedResources, analyzers.Resource{
+			Name:               user.FullName,
+			FullyQualifiedName: user.FullName,
+			Type:               "user",
+			Metadata: map[string]any{
+				"username": user.Username,
+				"role":     user.Role.Name,
+			},
+		})
+	}
```

Comment:
This is appending to `result.UnboundedResources`, which was allocated as a slice of non-zero length above: `make([]analyzers.Resource, len(info.Users))`.

Either change the assignment here to use the index:
```go
for i, user := range info.Users {
    result.UnboundedResources[i] = analyzers.Resource{
        // ...
    }
}
```

or change the initial assignment to allocate capacity:
```go
make([]analyzers.Resource, 0, len(info.Users))
```

Reply:
Nice catch 🙇